_posts/2018-11-20-mongodb-europe-018.md

```diff
@@ -0,0 +1,411 @@
+
+### Key takeaways from this session
+
+Generally speaking, there are three major causes of query slowness:
+- Blocking operations
+- Using the `$or` operator
+- Case insensitive searches
+
+It's not uncommon that a properly tuned query delivers a factor 1000 speed-up.
+So it's definitely worth investigating.
+
+### Problem 1: blocking operations
+This happens when you use an operator that needs all the data before producing results, so results can't be streamed.
+The most common culprits are the aggregation operators such as `$sort`, `$group`, `$bucket`, `$count` and `$facet`.
+
+Possible solutions:
+- Create a compound index to support your query and make sure that the sort order in the index is the same as in your query.
+- Offload the query to a secondary member.
+- Work around the issue by using a precalculated count.
+
+### Problem 2: `$and` is fast, `$or` is slow
+Sometimes a query is fast when you use the `$and` operator but slow when you use the `$or` operator.
+
+Solution:
+- Use a compound index to support `$and` queries.
+- Use separate single field indexes to support `$or` queries.
+
+### Problem 3: case insensitive searches are slow!
+It is much harder for MongoDB to perform case insensitive searches because it has to match all possible permutations of the search string. 
+For example, if you do a case insensitive search for the string "ace", it has to match "ace","Ace","aCe","ACe", and so on...
+
+Solution:
+- (3.4 and higher) Support the query with a case insensitive index.
+- Alternatively, store a toLower() copy of the string in another field and index and query that field instead.
+
+### General tips and tricks
+* Create an index on an element you are interested in instead of scanning the whole table.
+* When you query on a combination of fields create a compound index for these fields and not separate indices on each field.
+* ...but be careful with the usage of `$or`!
+* Build indices in the background instead of making it a blocking operation.
+* Do not index all fields as this will negatively impact write performance. 
+Investigate what you really need!
+* Use `.explain()` to analyze queries.
+* Ops Manager and Atlas have a [Performance Advisor](https://docs.atlas.mongodb.com/performance-advisor/) to help you identify problematic queries.
+* Train your people.
+* Work smarter, not harder!
+
+# Stitch
+([Ch-Ch-Ch-Ch-Changes: Taking Your MongoDB Stitch Application to the Next Level With Triggers, Andrew Morgan](https://sched.co/FmAJ){:target="_blank" rel="noopener noreferrer"})
+> Write less code and build apps faster! 
+
+Stitch is the 'Serverless platform from MongoDB', and it comes with a free tier to play around!
+Stitch provides a very easy way to implement new functionality without having to write lots of code in a separate backend application.
+The functionalities of Stitch are provided through an SDK.
```

Comment:
3rd Stitch and 2nd functionality in a row. Try to rewrite this sentence and avoid using these words. Since both this and the next one are quite short, perhaps combine them into 1?